### PR TITLE
[rancher v2.9] update docs for rancher ui extensions

### PR DESCRIPTION
**Pull request by diogoasouza**
## Description

This pull request adds changes to the rancher extension documentation to reflect the changes that will happen with the release of rancher v2.9.0.

More information can be found in the Jira issue.

### CODE REVIEW

File: docs/integrations-in-rancher/rancher-extensions.md [hunk 1]
```diff
@@ -14,21 +14,35 @@ Examples of built-in Rancher extensions are Fleet, Explorer, and Harvester. Exam
...
+3. When enabled for the second time forwards, it will expose the required endpoints again and, if needed, create any new CRD. The extensions that were already installed will be loaded after the rancher pod restarts.
 
-2. If not already installed in **Apps**, you must enable the extension operator by clicking the **Enable** button.
+### Caching extension files
 
-    - Click **OK** to add the Rancher extension repository if your installation is not air-gapped. Otherwise, uncheck the box to do so and click **OK**.
+By default, Rancher will cache every extension file in the filesystem. You can change that behavior by changing the value of `plugin.noCache` to `true`.
 
-    ![Rancher extension repository](/img/add-rancher-extension-repo.png)
+Rancher, however, limits the size of files cached to 20MB, and in case an extension has a file bigger than that,
```
Comment (martyav) [inline docs/integrations-in-rancher/rancher-extensions.md:35]:
```suggestion
Rancher does have a cached file size limit of 20MB. If an extension has a file bigger than that,
```

File: docs/integrations-in-rancher/rancher-extensions.md [hunk 2]
```diff
@@ -14,21 +14,35 @@ Examples of built-in Rancher extensions are Fleet, Explorer, and Harvester. Exam
...
-4. If no extensions are showing as available, you may manually add repos as follows:
+## Installing Extensions
+
+1. Click **☰ > Extensions** under **Configuration**.
+
+2. On the **Extensions** page, click on the **Available** tab to select which extensions you want to install.
+
+3. If no extensions are showing as available, you may manually add repos as follows:
 
     4.1. On the upper right of screen, click on **⋮ > Manage Repositories > Create**.
```
Comment (martyav) [inline docs/integrations-in-rancher/rancher-extensions.md:47]:
```suggestion
    4.1. On the upper right, click **⋮ > Manage Repositories > Create**.
```

File: docs/integrations-in-rancher/rancher-extensions.md [hunk 3]
```diff
@@ -38,11 +52,11 @@ Examples of built-in Rancher extensions are Fleet, Explorer, and Harvester. Exam
 
     ![Manage repositories](/img/manage-repos.png)
 
-5. Under the **Available** tab, click **Install** on the desired extension and version as in the example below. You can also update your extension from this screen, as the button to **Update** will appear on the extension if one is available.
+4. Under the **Available** tab, click **Install** on the desired extension and version as in the example below. You can also update your extension from this screen, as the button to **Update** will appear on the extension if one is available.
```
Comment (martyav) [inline docs/integrations-in-rancher/rancher-extensions.md:55]:
```suggestion
4. Under the **Available** tab, click **Install** on the desired extension and version as in the example below. You can also update your extension from this screen, as the button to **Update** appears on the extension if an update is available.
```

File: docs/integrations-in-rancher/rancher-extensions.md [hunk 4]
```diff
@@ -38,11 +52,11 @@ Examples of built-in Rancher extensions are Fleet, Explorer, and Harvester. Exam
...
 
     ![Install Kubewarden](/img/install-kubewarden.png)
 
-6. Click the **Reload** page button that will appear after your extension successfully installs. Note that a logged-in user who has just installed an extension will not see a change to the UI **unless** they reload the page.
+5. Click the **Reload** page button that will appear after your extension successfully installs. Note that a logged-in user who has just installed an extension will not see a change to the UI **unless** they reload the page.
```
Comment (martyav) [inline docs/integrations-in-rancher/rancher-extensions.md:59]:
```suggestion
5. Click the **Reload** button after your extension successfully installs to check its status. Updates to the UI aren't visible until you reload the page.
```

File: docs/integrations-in-rancher/rancher-extensions.md [hunk 5]
```diff
@@ -91,6 +105,15 @@ You must reload the page after disabling extensions or display issues may occur.
 
 :::
 
+## Enabling unauthenticated access to an extension
```
Comment (martyav) [inline docs/integrations-in-rancher/rancher-extensions.md:108]:
```suggestion
## Enabling Unauthenticated Access to an Extension
```

**Review by martyav — CHANGES_REQUESTED** [review]

**Comment by diogoasouza** [comment]
@martyav Thanks for the review! I updated the PR

File: docs/integrations-in-rancher/rancher-extensions.md [hunk 6]
```diff
@@ -14,35 +14,48 @@ Examples of built-in Rancher extensions are Fleet, Explorer, and Harvester. Exam
...
 
-> You must log in as an admin in order to view and interact with the extensions management page.
+> 1. You must log in as an administrator to view and interact with the extensions management page.
+> 2. You must enable extension support.
 
-## Installing Extensions
+## Enabling Extension Support in Rancher
 
-1. Click **☰ > Extensions** under **Configuration**.
+In Rancher v2.9.0 and later, extension support is now included in Rancher Manager by default.
```
Comment (martyav) [inline docs/integrations-in-rancher/rancher-extensions.md:22]:
```suggestion
In Rancher v2.9.0 and later, extension support is now included by default.
```

File: docs/integrations-in-rancher/rancher-extensions.md [hunk 7]
```diff
@@ -14,35 +14,48 @@ Examples of built-in Rancher extensions are Fleet, Explorer, and Harvester. Exam
...
+
+You can confirm if extension support is enabled by checking the `uiextension` feature flag. Any changes to this feature flag cause the Rancher pod to restart.
+
+When extension support is enabled for the first time, it creates resources, such as the CRD, so that UI Extensions can work.
+When extension support is disabled, it disables the endpoints and does not cache any files. However, it does not remove any CRs or delete any extensions that were installed before.
+If re-enabled, it exposes the required endpoints again and creates CRDs as needed. The extensions that were already installed load after the Rancher pod restarts.
```
Comment (martyav) [inline docs/integrations-in-rancher/rancher-extensions.md:28]:
```suggestion
When you enable extension support for the first time, it creates resources, such as the CRDs, so that UI Extensions can work. When extension support is disabled, it disables the endpoints and does not cache any files. However, it does not remove any CRs or delete any extensions that were installed before. If re-enabled, it exposes the required endpoints again and creates CRDs as needed. The extensions that were already installed load after the Rancher pod restarts.
```

File: docs/integrations-in-rancher/rancher-extensions.md [hunk 8]
```diff
@@ -14,35 +14,48 @@ Examples of built-in Rancher extensions are Fleet, Explorer, and Harvester. Exam
...
 
-2. If not already installed in **Apps**, you must enable the extension operator by clicking the **Enable** button.
+### Caching Extension Files
 
-    - Click **OK** to add the Rancher extension repository if your installation is not air-gapped. Otherwise, uncheck the box to do so and click **OK**.
+By default, Rancher caches every extension file in the filesystem. You can change that behavior by setting `plugin.noCache` to `true`.
```
Comment (martyav) [inline docs/integrations-in-rancher/rancher-extensions.md:32]:
```suggestion
By default, Rancher caches every extension file in the file system. You can change that behavior by setting `plugin.noCache` to `true`.
```

File: docs/integrations-in-rancher/rancher-extensions.md [hunk 9]
```diff
@@ -14,35 +14,48 @@ Examples of built-in Rancher extensions are Fleet, Explorer, and Harvester. Exam
...
 
-    ![Rancher extension repository](/img/add-rancher-extension-repo.png)
+Rancher does have a cached file size limit of 20MB. If an extension has a file bigger than that,
+the cache is disabled and `plugin.noCache` is set to `true`, regardless of user input.
```
Comment (martyav) [inline docs/integrations-in-rancher/rancher-extensions.md:35]:
```suggestion
Rancher does have a cached file size limit of 20MB. If an extension has a file bigger than that, the cache is disabled and `plugin.noCache` is set to `true`, regardless of user input.
```

File: docs/integrations-in-rancher/rancher-extensions.md [hunk 10]
```diff
@@ -14,35 +14,48 @@ Examples of built-in Rancher extensions are Fleet, Explorer, and Harvester. Exam
...
 
-3. On the **Extensions** page, click on the **Available** tab to select which extensions you want to install.
 
-4. If no extensions are showing as available, you may manually add repos as follows:
+## Installing Extensions
+
+1. Click **☰ > Extensions** under **Configuration**.
 
-    4.1. On the upper right of screen, click on **⋮ > Manage Repositories > Create**.
+2. On the **Extensions** page, select the **Available** tab to choose extensions you want to install.
```
Comment (martyav) [inline docs/integrations-in-rancher/rancher-extensions.md:42]:
```suggestion
2. On the **Extensions** page, select the **Available** tab to choose the extensions that you want to install.
```

File: docs/integrations-in-rancher/rancher-extensions.md [hunk 11]
```diff
@@ -14,35 +14,48 @@ Examples of built-in Rancher extensions are Fleet, Explorer, and Harvester. Exam
...
+
+3. If no extensions are showing as available, you may manually add repos as follows:
```
Comment (martyav) [inline docs/integrations-in-rancher/rancher-extensions.md:44]:
```suggestion
3. If no extensions are listed as available, you can manually add the repos:
```

File: docs/integrations-in-rancher/rancher-extensions.md [hunk 12]
```diff
@@ -14,35 +14,48 @@ Examples of built-in Rancher extensions are Fleet, Explorer, and Harvester. Exam
...
+
+    4.1. On the upper right, click **⋮ > Manage Repositories > Create**.
 
     4.2. Add the desired repo name, making sure to also specify the Git Repo URL and the Git Branch.
```
Comment (martyav) [inline docs/integrations-in-rancher/rancher-extensions.md:48]:
```suggestion
    4.2. Add the desired repo name, making sure to also specify the Git repo URL and the Git branch.
```

Not part of the PR but weird capitalization

File: docs/integrations-in-rancher/rancher-extensions.md [hunk 13]
```diff
@@ -14,35 +14,48 @@ Examples of built-in Rancher extensions are Fleet, Explorer, and Harvester. Exam
...
 
     4.3. Click **Create** in the lower right again to complete.
 
     ![Manage repositories](/img/manage-repos.png)
 
-5. Under the **Available** tab, click **Install** on the desired extension and version as in the example below. You can also update your extension from this screen, as the button to **Update** will appear on the extension if one is available.
+4. Under the **Available** tab, click **Install** on the desired extension and version as in the example below. You can also update your extension from this screen, as the button to **Update** appears on the extension if an update is available.
```
Comment (martyav) [inline docs/integrations-in-rancher/rancher-extensions.md:54]:
```suggestion
4. Under the **Available** tab, click **Install** on the desired extension and version, as in the example below. You can also update your extension from this screen. The button to **Update** appears on the extension card if an update is available.
```

File: docs/integrations-in-rancher/rancher-extensions.md [hunk 14]
```diff
@@ -14,35 +14,48 @@ Examples of built-in Rancher extensions are Fleet, Explorer, and Harvester. Exam
...
 
     ![Install Kubewarden](/img/install-kubewarden.png)
 
-6. Click the **Reload** page button that will appear after your extension successfully installs. Note that a logged-in user who has just installed an extension will not see a change to the UI **unless** they reload the page.
+5. Click the **Reload** button after your extension successfully installs to check its status. Updates to the UI aren't visible until you reload the page.
```
Comment (martyav) [inline docs/integrations-in-rancher/rancher-extensions.md:58]:
```suggestion
5. Click **Reload** after your extension successfully installs to check its status. Updates to the UI aren't visible until you reload the page.
```

**Review by martyav — CHANGES_REQUESTED** [review]

**Comment by diogoasouza** [comment]
@martyav Updated the PR, thanks

File: docs/integrations-in-rancher/rancher-extensions.md [hunk 15]
```diff
@@ -14,35 +14,45 @@ Examples of built-in Rancher extensions are Fleet, Explorer, and Harvester. Exam
...
 
-> You must log in as an admin in order to view and interact with the extensions management page.
+> 1. You must log in as an administrator to view and interact with the extensions management page.
+> 2. You must enable extension support.
 
-## Installing Extensions
+## Enabling Extension Support in Rancher
 
-1. Click **☰ > Extensions** under **Configuration**.
+In Rancher v2.9.0 and later, extension support is now included by default.
```
Comment (martyav) [inline docs/integrations-in-rancher/rancher-extensions.md:22]:
```suggestion
Rancher v2.9.0 and later includes extension support.
```

File: docs/integrations-in-rancher/rancher-extensions.md [hunk 16]
```diff
@@ -14,35 +14,45 @@ Examples of built-in Rancher extensions are Fleet, Explorer, and Harvester. Exam
...
+2. On the **Extensions** page, select the **Available** tab to choose the extensions that you want to install.
 
-    4.2. Add the desired repo name, making sure to also specify the Git Repo URL and the Git Branch.
+3. If no extensions are listed as available, you can manually add the repos:
+
+    4.1. On the upper right, click **⋮ > Manage Repositories > Create**.
+
+    4.2. Add the desired repo name, making sure to also specify the Git repo URL and the Git branch.
 
     4.3. Click **Create** in the lower right again to complete.
```
Comment (martyav) [inline docs/integrations-in-rancher/rancher-extensions.md:47]:
```suggestion
    3.1. On the upper right, click **⋮ > Manage Repositories > Create**.

    3.2. Add the desired repo name, making sure to also specify the Git repo URL and the Git branch.

    3.3. Click **Create** in the lower right again to complete.
```

**Review by martyav — CHANGES_REQUESTED** [review]

**Comment by diogoasouza** [comment]
@martyav Done!

File: docs/integrations-in-rancher/rancher-extensions.md [hunk 17]
```diff
@@ -91,6 +101,13 @@ You must reload the page after disabling extensions or display issues may occur.
 
 :::
 
+## Enabling Unauthenticated Access to an Extension
+
+In Rancher v2.9.0, you can allow unauthenticated access to an extension. You may want to enable unauthenticated access if the extension enables a new locale or adds custom branding. By default, all extensions require user authentication to load. 
+
+To enable unauthenticated access to an extension, set `plugin.noAuth` to `true` in the CR used by the extension.
+
+
```
Comment (recena) [inline docs/integrations-in-rancher/rancher-extensions.md:110]: @diogoasouza Remove this extra empty line

**Comment by martyav** [comment]
@diogoasouza Before we merge, can you copy the file to versioned_docs/version-2.9/integrations-in-rancher/rancher-extensions.md and restore docs/integrations-in-rancher/rancher-extensions.md to the state it was in before the PR was opened? We recently added a preview section for v2.9 documentation on the live site. When PRs are merged, they're automatically published, so having the file in  versioned_docs/version-2.9 allows us to safely add v2.9 files until v2.9 is released and becomes the new Latest version.

File: docs/integrations-in-rancher/rancher-extensions.md (hunk 17 above)
Comment (diogoasouza) [reply docs/integrations-in-rancher/rancher-extensions.md]: Done

**Comment by diogoasouza** [comment]
@martyav Got it! Updated the PR

**Review by martyav — APPROVED** [review]

**Comment by diogoasouza** [comment]
@martyav Thanks for the approval. When can we have this merged? I don't have write permission

**Comment by diogoasouza** [comment]
@martyav they were addressed
